internal/cmd/workspaces.go

```diff
@@ -85,15 +86,25 @@ func lsWorkspacesCommand() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			workspaces, err := getWorkspaces(ctx, client, user)
-			if err != nil {
-				return err
-			}
-			if provider != "" {
+			var workspaces []coder.Workspace
+			if all {
+				var err error
+				workspaces, err = getAllWorkspaces(ctx, client)
+				if err != nil {
+					return err
+				}
+			} else if provider != "" {
+				var err error
 				workspaces, err = getWorkspacesByProvider(ctx, client, provider, user)
 				if err != nil {
 					return err
 				}
+			} else {
+				var err error
+				workspaces, err = getWorkspaces(ctx, client, user)
+				if err != nil {
+					return err
+				}
 			}
```

Comment:
Shuffled this around because it seemed like the first fetch for the user-only workspaces was ignored if providers was also passed.